src/Illuminate/Session/SessionManager.php

```diff
@@ -125,7 +126,9 @@ protected function createWincacheDriver()
      */
     protected function createRedisDriver()
     {
-        $handler = $this->createCacheHandler('redis');
+        $minutes = $this->app['config']['session.lifetime'];
+        $cache = new CacheManager($this->app);
```

Comment:
I pushed a new commit. 
`$cache = clone $this->app['cache'];` before `setConnection`, is this acceptable?

$this->app['cache'] is registered as a singleton, I can't think a better way than clone it.